src/deroff.rs

```diff
@@ -376,6 +376,13 @@ impl Deroffer {
         }
     }
 
+    fn digit<'a>(s: &'a str, idx: usize) -> bool {
```

Comment:
Lifetime annotation can be elided here.
```suggestion
    fn digit(s: &str, idx: usize) -> bool {
```